web/packages/teleterm/src/ui/services/workspacesService/accessRequestsService.ts

```diff
@@ -54,42 +60,219 @@ export class AccessRequestsService {
     });
   }
 
-  getAddedResourceCount() {
+  getAddedResourceCount(): number {
     const pendingAccessRequest = this.getState().pending;
-    return (
-      Object.keys(pendingAccessRequest.node).length +
-      Object.keys(pendingAccessRequest.db).length +
-      Object.keys(pendingAccessRequest.app).length +
-      Object.keys(pendingAccessRequest.kube_cluster).length +
-      Object.keys(pendingAccessRequest.windows_desktop).length +
-      Object.keys(pendingAccessRequest.user_group).length
-    );
+    if (pendingAccessRequest.kind === 'role') {
+      return pendingAccessRequest.roles.size;
+    }
+    if (pendingAccessRequest.kind === 'resource') {
+      return pendingAccessRequest.resources.size;
+    }
```

Comment:
This could also be a switch with a default branch with `satisfies never` to add some extra type safety.